File: Mu2e-STMDAQ/OLD_BUT_DONT_DELETE_YET/NOT_NEEDED/hardwareScripts/repeat.cc
///////////////////////////////////////////////////////////
// Repeat module: interface with axi_repeat star (main) 
///////////////////////////////////////////////////////////

// repeat
#include "STMDAQ-TestBeam/hardwareScripts/repeat.hh"

//Standard constructor - shouldn't be used
repeat::repeat() {}

// Set the burst size in axi_repeat size 
int repeat::setburstlength(IPBusManager* hw, int repeatnode_nr, int burstlength){

  std::string node ="Repeats.repeat_"+std::to_string(repeatnode_nr)+".burst_size";
  hw->write(node,burstlength);

  return 1;
  
}

// Prepare the wave form memory (WFM) for getting data     
int repeat::prepare_wfm_upload(IPBusManager* hw, int repeatnode_nr){

  std::string node ="Repeats.repeat_"+std::to_string(repeatnode_nr)+".ctrl";
  hw->write(node,0x08);

  return 1;
  
}

// Arm the unit, start to play the WFM data  
int repeat::arm(IPBusManager* hw, int repeatnode_nr){

  std::string node ="Repeats.repeat_"+std::to_string(repeatnode_nr)+".ctrl";
  hw->write(node,0x01);

  return 1;
  
}

// Disarm the unit 
int repeat::disarm(IPBusManager* hw, int repeatnode_nr){

  std::string node ="Repeats.repeat_"+std::to_string(repeatnode_nr)+".ctrl";
  hw->write(node,0x02);

  return 1;
  
}

// Enable external trigger in the unit   
int repeat::rst(IPBusManager* hw, int repeatnode_nr){

  std::string node ="Repeats.repeat_"+std::to_string(repeatnode_nr)+".ctrl";
  hw->write(node,0x10);

  return 1;
  
}

// Check repeat node status
uint64_t repeat::check_status(IPBusManager* hw, int repeatnode_nr){

  std::string node ="Repeats.repeat_"+std::to_string(repeatnode_nr)+".status";
  uint64_t value = hw->read(node);

  return value;
  
}

File: Mu2e-STMDAQ/OLD_BUT_DONT_DELETE_YET/NOT_NEEDED/hardwareScripts/FMC144/pythonSource/repeat.py
#!/usr/bin/python

# -*- coding: utf-8 -*-
import sys
import time
import uhal
import spi_functions as s

def repeat_setburstlength (hw,repeatnode_nr,burstlength):
  
  hw.getNode("Repeats.repeat_%s.burst_size" %(repeatnode_nr)).write(burstlength)
  hw.dispatch()
  
  return 1

def repeat_prepare_wfm_upload (hw,repeatnode_nr):
  hw.getNode("Repeats.repeat_%s.ctrl" %(repeatnode_nr)).write(0x08)
  hw.dispatch()
  
  return 1
  
def repeat_arm(hw,repeatnode_nr):
  hw.getNode("Repeats.repeat_%s.ctrl" %(repeatnode_nr)).write(0x01)
  hw.dispatch()
  
  return 1

def repeat_disarm(hw,repeatnode_nr):
  hw.getNode("Repeats.repeat_%s.ctrl" %(repeatnode_nr)).write(0x02)
  hw.dispatch()
  
  return 1

def repeat_rst(hw,repeatnode_nr):
  hw.getNode("Repeats.repeat_%s.ctrl" %(repeatnode_nr)).write(0x10)
  hw.dispatch()
  
  return 1
  
def repeat_check_status(hw,repeatnode_nr):
  value = hw.getNode("Repeats.repeat_%s.status" %(repeatnode_nr)).read()
  hw.dispatch()
  
  return value
